src/components/hoc/createPlayerInfo.tsx

```diff
@@ -0,0 +1,51 @@
+import MageHero from '../../classes/MageHero.tsx';
+import MeleeHero from '../../classes/MeleHero.tsx';
+import RangeHero from '../../classes/RangeHero.tsx';
+import savePlayer from '../../functions/heroFunctions/savePlayer.tsx'
+import Button from '../button.tsx';
+import { defaultHero } from '../../typesAndConstants/constants.tsx'
+
+type DefaultHeroSerializer = (playerName: string) => HeroInfo;
+const createHeroMap = new Map<HERO_TYPES, DefaultHeroSerializer>([
+    ["Mage", MageHero.toJSON],
+    ["Melee", MeleeHero.toJSON],
+    ["Range", RangeHero.toJSON]
+]);
+
+export default function createPlayer(WrappedComponent: typeof Button, type: HERO_TYPES, playerName: string, callbackFunction: React.Dispatch<React.SetStateAction<HeroInfo>>, index?: number) {
+    const content = (
+        <div>{type}</div>
+    )
```

Comment:
If type is string do you need to wrap it in div

Reply:
its for css purpose